service/matching/matching_engine.go

```diff
@@ -1824,7 +1824,9 @@ func (e *matchingEngineImpl) unloadTaskQueuePartitionByKey(partition tqid.Partit
 	foundTQM, ok := e.partitions[key]
 	if !ok || (unloadPM != nil && foundTQM != unloadPM) {
 		e.partitionsLock.Unlock()
-		unloadPM.Stop()
+		if unloadPM != nil {
+			unloadPM.Stop()
```

Comment:
I don't like how some of this stuff changed in the branch.  I forgot if I brought this up in code review, but I don't think `getTaskQueuePartitionManager` should call `unloadTaskQueuePartition` if `WaitUntilInitialized` fails. Mainly, because we don't control the context timeout, and if the caller (i.e. external client) sets a very short deadline, we might get an error here because that deadline times out even though everything is actually fine. I think if we get errors during load, the partition manager should unload itself (which it already does).

If we do that, then we don't need to `unloadPM.Stop()` here at all.

Reply:
Removing Stop from here causes some problems that needs further discussion.